### PR TITLE
Always deploy xcodebuild archive and exportArchive command logs.

### PR DESCRIPTION
**Pull request by godrei**
### Checklist

- [x] I've read and accepted the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- Requires _MINOR_ [version update](https://semver.org/)

### Context

Xcode Archive step is not exporting the raw xcodebuild logs, in case the step succeeds. This makes debugging an issue (which does not make the step failing) impossible, as the step prints only the last few lines of the xcodebuild log.

Resolves: https://bitrise.atlassian.net/browse/STEP-1042

### Changes

- differentiate `xcodebuild archive` and `xcodebuild -exportArchive` logs (`BITRISE_XCODEBUILD_ARCHIVE_LOG_PATH` and `BITRISE_XCODEBUILD_EXPORT_ARCHIVE_LOG_PATH`)
- deploy both `xcodebuild archive` and `xcodebuild -exportArchive` logs regardless the step run status
- create check-output step to validate the Step Outputs

### CODE REVIEW

File: _scripts/check-output/main.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
+package main
```
Comment (mateherber) [inline _scripts/check-output/main.go:1]: Don't we want to put this functionality to the `steps-check` step?

File: step.yml [hunk 2]
```diff
@@ -360,13 +360,15 @@ outputs:
...
-  - BITRISE_XCODE_ARCHIVE_RAW_RESULT_TEXT_PATH:
-    opts:
-      title: The full, raw archive output file path
-      description: |-
-        This is the path of the raw archive results log file.
 
-        If `output_tool=xcpretty` and the archive fails this log will contain the archive output.
+  - BITRISE_XCODEBUILD_ARCHIVE_LOG_PATH:
+    opts:
+      title: "`xcodebuild archive` command log file path"
```
Comment (mateherber) [inline step.yml:366]: `summary` or `description` would be nice for all inputs and outputs

File: _scripts/check-output/main.go (hunk 1 above)
Comment (godrei) [reply _scripts/check-output/main.go:1]: I think it is more like our `simple-git-clone` step, a utility Step we can use in the Step e2e test cases.
I suggest moving it to a separate repository.

File: step.yml (hunk 2 above)
Comment (godrei) [reply step.yml:366]: Agree, unfortunately only the Title is presented in the WFEditor, and in the case of Step Inputs, only the Title property is required: https://github.com/bitrise-io/stepman/blob/203f7a48d37aebc3cc5211690e9365f4d6d98d72/models/models_methods.go#L121-L162.

I will make up the missing description for the new outputs.

File: _scripts/check-output/main.go (hunk 1 above)
Comment (godrei) [reply _scripts/check-output/main.go:1]: https://github.com/bitrise-steplib/bitrise-step-check-step-outputs

File: step.yml (hunk 2 above)
Comment (godrei) [reply step.yml:366]: Description added in https://github.com/bitrise-steplib/steps-xcode-archive/pull/238/commits/1d12da3b171083fc813e01d1f51aca6f63c8cc4d

**Review by lpusok — APPROVED** [review]

**Pull request merged**